src/System.Private.CoreLib/shared/System/Number.NumberToDouble.cs

```diff
@@ -0,0 +1,449 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System
+{
+    internal unsafe partial class Number
+    {
+        // precomputed tables with powers of 10. These allows us to do at most
+        // two Mul64 during the conversion. This is important not only
+        // for speed, but also for precision because of Mul64 computes with 1 bit error.
+
+        private static readonly ulong[] rgval64Power10 = new ulong[]
```

Comment:
Nit: These should have `s_` prefix

Reply:
And you can omit `rg` (`rg` is hungarian notation for array).

Reply:
Fixed.